rcgen/src/lib.rs

```diff
@@ -32,37 +32,46 @@ println!("{}", key_pair.serialize_pem());
 #![deny(missing_docs)]
 #![allow(clippy::complexity, clippy::style, clippy::pedantic)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![warn(unreachable_pub)]
 
-#[cfg(feature = "crypto")]
-use ring_like::digest;
 use std::collections::HashMap;
 use std::fmt;
 use std::hash::Hash;
 use std::net::IpAddr;
 #[cfg(feature = "x509-parser")]
 use std::net::{Ipv4Addr, Ipv6Addr};
+
 use time::{OffsetDateTime, Time};
 use yasna::models::ObjectIdentifier;
 use yasna::models::{GeneralizedTime, UTCTime};
 use yasna::tags::{TAG_BMPSTRING, TAG_TELETEXSTRING, TAG_UNIVERSALSTRING};
 use yasna::DERWriter;
 use yasna::Tag;
 
-pub use crate::certificate::{
+mod certificate;
+pub use certificate::{
 	date_time_ymd, BasicConstraints, Certificate, CertificateParams, CidrSubnet, CustomExtension,
 	DnType, ExtendedKeyUsagePurpose, GeneralSubtree, IsCa, NameConstraints,
 };
-pub use crate::crl::{
+mod crl;
+pub use crl::{
 	CertificateRevocationList, CertificateRevocationListParams, CrlDistributionPoint,
 	CrlIssuingDistributionPoint, CrlScope, RevocationReason, RevokedCertParams,
 };
-pub use crate::csr::{CertificateSigningRequestParams, PublicKey};
-pub use crate::error::{Error, InvalidAsn1String};
-use crate::key_pair::PublicKeyData;
-pub use crate::key_pair::{KeyPair, RemoteKeyPair};
-use crate::oid::*;
+mod csr;
```

Comment:
I think it's nicer before the reorderings.

Reply:
Changed this so it only moves the `mod` declarations up and removes the unnecessary `crate::` prefixes from the imports.

Reply:
> separating use from mod?

Yeah that separation is useful I think.

Reply:
So do you want to approve this in its current state?

Reply:
Done, approved.